crates/autopilot/src/run_loop.rs

```diff
@@ -36,19 +39,21 @@ use {
         sync::Arc,
         time::{Duration, Instant},
     },
-    tracing::{warn, Instrument},
+    tokio::sync::{mpsc, mpsc::UnboundedSender, Mutex},
+    tracing::Instrument,
 };
 
 pub struct RunLoop {
     pub eth: infra::Ethereum,
     pub persistence: infra::Persistence,
-    pub drivers: Vec<infra::Driver>,
+    pub drivers: Vec<Arc<infra::Driver>>,
```

Comment:
Last nit:
I think with the introduction of `RunLoop::new()` all these fields no longer have to be `pub`.